llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -8833,6 +8852,8 @@ class BoUpSLP::ShuffleCostEstimator : public BaseShuffleAnalysis {
     } else if (!V1 && V2) {
       // Shuffle vector and tree node.
       unsigned VF = cast<FixedVectorType>(V2->getType())->getNumElements();
+      if (auto *VecTy = dyn_cast<FixedVectorType>(ScalarTy))
+        VF /= VecTy->getNumElements();
```

Comment:
assertion. Better to have a function, doing this, to avoid duplication

Reply:
Add assertion in `GetVF`.